### PR TITLE
Ports Dolphins

### PR DESCRIPTION
**Pull request by Altoids1**
### Fuck yeah dolphins
The most requested port by far for sure
dolphin2

#### Changelog

:cl:  
rscadd: Dolphins are back!!
/:cl:

### CODE REVIEW

File: yogstation/code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	spacewalk = TRUE
+
+	environment_smash = 0
+	harm_intent_damage = 8
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	pass_flags = PASSTABLE //secretly a tactical dolphin buff.
+	attacktext = "bites"
+	attack_sound = 'sound/weapons/bite.ogg'
+	speak_emote = list("Chitters", "Squeeks", "Clicks")
```
Comment (GuppyLaxx) [inline yogstation/code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: This shouldn't be capitalized

**Review by GuppyLaxx — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+	//Space dolphins aren't affected by cold.
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 1500
+
+	faction = list("dolphin")//why do carps not attack dolphins again? //i dunno now they do
+
+/mob/living/simple_animal/hostile/retaliate/dolphin/Process_Spacemove(movement_dir = 0)
+	return 1	//No drifting in space for space dolphins!	//original comments do not steal //Too late.
```
Comment (ThatLing) [inline yogstation/code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: return TRUE

File: yogstation/code/modules/events/dolphin_migration.dm [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+/datum/round_event/dolphin_migration/setup()
+	startWhen = rand(40, 60)
+
+/datum/round_event/dolphin_migration/announce()
+	priority_announce("Unknown biological entities have been detected near [station_name()], please stand-by.", "Lifesign Alert")
+
+
+/datum/round_event/dolphin_migration/start()
+	for(var/obj/effect/landmark/carpspawn/C in GLOB.landmarks_list)
+		new /mob/living/simple_animal/hostile/retaliate/dolphin(C.loc)
```
Comment (ThatLing) [inline yogstation/code/modules/events/dolphin_migration.dm:22]: Spawn a dolphin on every carpspawn landmark? That's a lot of dolphins.

File: yogstation/code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm (hunk 2 above)
Comment (Time-Green) [inline yogstation/code/modules/mob/living/simple_animal/hostile/retaliate/dolphin.dm]: youre fucking dead kiddo

File: yogstation/code/modules/events/dolphin_migration.dm (hunk 3 above)
Comment (Altoids1) [reply yogstation/code/modules/events/dolphin_migration.dm:22]: Compare that to what the carp_migration event does:
```dm
/datum/round_event/carp_migration/start()
	for(var/obj/effect/landmark/carpspawn/C in GLOB.landmarks_list)
		if(prob(95))
			new /mob/living/simple_animal/hostile/carp(C.loc)
		else
			new /mob/living/simple_animal/hostile/carp/megacarp(C.loc)
```

**Comment by Altoids1** [comment]
So I recently received some new dolphin sprites from Bawhoppen over on Discord.
dolphin5
dolphin4
Do y'all think these are better? Should I swap the old sprites for these ones?

File: yogstation/code/modules/events/dolphin_migration.dm (hunk 3 above)
Comment (ThatLing) [reply yogstation/code/modules/events/dolphin_migration.dm:22]: Fair enough

**Comment by ThatLing** [comment]
@Altoids1 Ready for merge or nah?

**Comment by Altoids1** [comment]
@ThatLing Ya ❤️

**Pull request merged**